magma/array.py

```diff
@@ -724,10 +725,6 @@ def _filter(item):
         assert len(overlaps) == 1
         key, value = overlaps[0]
         self._remove_slice(key)
-        if value._ts:
-            self._ts[index] = value[index - key[0]]
-        else:
-            value._ts[index - key[0]] = self._ts[index] = self._make_t(index)
```

Comment:
This is redundant with the `_resolve_slice_children` logic, we don't need to duplicate it here.